### ISSUE TITLE
Branch Strategies don`t work

### ISSUE THREAD

**Issue by melodous** [jenkinsci/seed-plugin#15]
I'm not able of make the "branch strategies" work. I'd need to disable update events on one specific branch but seem seed plugin doesn't follow the configuration options, it still launch pipeline seed and build.

Jenkins version 1.620 seed version 0.14.3

Configuration:

```yaml
extensions:
  - id: ontrack
    dsl: |
      configure { node ->
          node / 'builders' / 'net.nemerosa.ontrack.jenkins.OntrackDSLStep' {
             'usingText' true
             'scriptText' "ontrack.branch(SEED_PROJECT, 'template').instance(SEED_BRANCH, [branch: BRANCH])"
             'injectEnvironment' 'SEED_PROJECT,SEED_BRANCH,BRANCH'
             'injectProperties' ''
             'ontrackLog' true
          }
      }

strategies:
  - id: no-update
    pipeline-trigger: false
    pipeline-auto: false

classes:
   - id: delta-cars
     branch-strategy: no-update
     pipeline-generator-extensions:
        - ontrack

projects:
   - id: insurancetelematics-drivies-cars-offset
     project-class: delta-cars
```

I also has been trying to attach branch-strategy configurations directly to the project with the same result.

**Comment by dcoraboeuf** [comment]
We are using custom strategies all the same at my client - so it must be a configuration problem. I'll test yours and come back to you. Not today though :(

**Comment by dcoraboeuf** [comment]
Hi Raul,

The pipeline parameters, `pipeline-trigger` and `pipeline-auto` in your case, are not parameters of the strategy (which defines a naming strategy) but of the project (or its class). You can adapt your global configuration the following way:

```yaml
extensions:
  - id: ontrack
    dsl: |
      configure { node ->
          node / 'builders' / 'net.nemerosa.ontrack.jenkins.OntrackDSLStep' {
             'usingText' true
             'scriptText' "ontrack.branch(SEED_PROJECT, 'template').instance(SEED_BRANCH, [branch: BRANCH])"
             'injectEnvironment' 'SEED_PROJECT,SEED_BRANCH,BRANCH'
             'injectProperties' ''
             'ontrackLog' true
          }
      }

classes:
   - id: no-update
     pipeline-trigger: false
     pipeline-auto: false
   - id: delta-cars
     project-class: no-update
     pipeline-generator-extensions:
        - ontrack

projects:
   - id: insurancetelematics-drivies-cars-offset
     project-class: delta-cars
```

Here, I've just use the possibility to inherit one class from another.

Note that a next version will allow to declare several classes, in order to combine them (see #7).

I'm not closing this issue yet, first waiting for your feedback, second because I have to clarify the documentation on this subject.

**Comment by melodous** [comment]
Sorry I can't make it work :( I see in the code that It should write on logs something like "Seed events are not enabled for project", but this doesn't happen.

An example of the new configuration, I also try inherit one class from another as you said :

classes:
  - id: ontrack
    pipeline-generator-extensions:
      - ontrack
  - id: drivies-cars-offset
    pipeline-trigger: false
    pipeline-auto: false
    pipeline-generator-extensions:
      - ontrack

projects:
   - id: insurancetelematics-drivies-cars-offset
     project-class: drivies-cars-offset
   - id: insurancetelematics-drivies-landing
     project-class: ontrack

And yes, I think it easy to misunderstood the location of the pipeline-trigger and pipeline-auto variables from documentation, maybe they should stay on other place, non on Branch-strategies's wiki page.

**Comment by dcoraboeuf** [comment]
Hi Raul,

I'll check again and use your configuration to create a test case (changing the names). I'll also adapt the documentation to make it clearer.

Thanks again.

**Comment by dcoraboeuf** [comment]
Just an additional remark: in your configuration, your `insurancetelematics-drivies-landing` project is not configured to have `pipeline-trigger` and `pipeline-auto` set to `false`. Is that correct?

**Comment by melodous** [comment]
Hi Damien, yes it's correct, in fact I just a relevant piece of configuration, we have more projects.

**Comment by dcoraboeuf** [comment]
On the `feature/15-config` branch, I've created two acceptance tests to test the `pipeline-auto` and `pipeline-trigger` flags in a class. No problem so far.

Just to be sure: you said you tried to make one class inherit from another, but I do not see this in your configuration sample:

```yaml
classes:
  - id: ontrack
    pipeline-generator-extensions:
      - ontrack
  - id: drivies-cars-offset
    pipeline-trigger: false
    pipeline-auto: false
    pipeline-generator-extensions:
      - ontrack

projects:
   - id: insurancetelematics-drivies-cars-offset
     project-class: drivies-cars-offset
   - id: insurancetelematics-drivies-landing
     project-class: ontrack
```

The `drivies-cars-offset` class does not inherit from the `ontrack` one (no `project-class` element), so the `insurancetelematics-drivies-cars-offset` has not the `pipeline-trigger` and `pipeline-auto` parameters set to `false`.

When does your problem occur? On both `insurancetelematics-drivies-cars-offset` and `insurancetelematics-drivies-landing` projects?

**Comment by dcoraboeuf** [comment]
Hi Raul, I still do not get your error... Is it possible to post your complete configuration file? Obfuscating some data of course?

**Comment by melodous** [comment]
Hi Damien, Sorry for the delay but I took holidays the last week and needed to finish some task before leaving, You know :). I'll check it in a few days.

**Comment by dcoraboeuf** [comment]
No problem. I'll probably merge the branch into the master though, just to get the additional acceptance tests.

**Comment by dcoraboeuf** [comment]
Hi, by finally being able to reproduce the problem elsewhere, I found an explanation. Stay tuned.

**Comment by dcoraboeuf** [comment]
Hi Raul, this is finally fixed. I have adapted the [documentation](https://github.com/jenkinsci/seed-plugin/wiki/Configuration-parameters) and the parameter you need is `pipeline-start-auto`. Note that your branch seed job needs to be regenerated.

**Comment by dcoraboeuf** [comment]
This will be available in version 0.16.

**Comment by melodous** [comment]
Hi Damien, great!. I just arrived from holidays this week I'll check this and other pending things (ontrack github enterprise integration) ASAP.
